### DISCUSSION TITLE
Using serial terminal on esp32 device

### DISCUSSION THREAD

**Discussion by PiotrHryciow** [arendst/Tasmota#17325] [Q&A] [answered]
Hi, at the start i want to point out that. I am new to the whole esp32 and tasmota topics. I succesfull installed Tasmota to my esp32 device, but now i want to unistall it. I read that it can be done using serial terminal program. But the problem is when i try using recomended PuTTY, and pairing with the device it prints out:

```
ets Jun  8 2016 00:22:57

rst:0x1 (POWERON_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
flash read err, 1000
ets_main.c 371
ets Jun  8 2016 00:22:57

rst:0x10 (RTCWDT_RTC_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0030,len:184
load:0x40078000,len:12588
ho 0 tail 12 room 4
load:0x40080400,len:2916
entry 0x400805c4

00:00:00.002 HDW: ESP32-D0WDQ6
00:00:00.058 UFS: FlashFS mounted with 308 kB free
00:00:00.129 CFG: Loaded from File, Count 7
00:00:00.144 QPC: Count 1
00:00:00.275 BRY: Berry initialized, RAM used=4041 bytes
00:00:00.303 Project tasmota - Tasmota Version 12.2.0(tasmota)-2_0_5(2022-10-17T                             08:39:59)
00:00:00.496 WIF: WifiManager active for 3 minutes
00:00:01.142 HTP: Web server active on tasmota-6E1268-4712 with IP address 192.1                             68.4.1
00:00:06.876 QPC: Reset
```
 
and i can't type anything in it. I tried running it for some time. but even after 10 minutes passed, nothing happened.
I also tried using diffrent programs:
- web tool for instaling tasmota (https://tasmota.github.io/install/),
- serial monitor in arduino app,
But in both cases now I can type something, but no matter what command I use. it always prints out:

`00:00:23.666 RSL: RESULT = {"Command":"Unknown"}`

I tried:
- restarting esp32 device,
- pluging it via different usb port on my computer,
- reinstalling tasmota using previous mentioned web tool

**Comment by barbudor** [comment, marked as answer]
Tasmota send logs for 10 minutes to the serial and then disable it. If you type any tasmota command, it will re-enable the serial logs
You get the error message because what you typed is not a proper tasmota command

You can't remove tasmota from the serial console using putty
Removing a firmware can only be done by erasing the flash. Generally you would do that before flashing another firmware. 
AFAIK the only software that would allow to erase the flash without necesseraly flashing another firmware would be esptool

**Reply by sfromis** [reply]
Yeah, in general, you do not "uninstall", but just flash something different on top, with erase included in the flashing process.

Failure to see something when typing tends to depend on terminal program settings, as Tasmota in half duplex mode will not echo typed chars (terminal program should have some local echo option), and hitting enter may not work, with ctrl-J being the "more basic" way of sending the new-line char.
